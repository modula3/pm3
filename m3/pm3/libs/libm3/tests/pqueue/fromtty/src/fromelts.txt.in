fromelts
showheap
fromelts e d c b a z y x w
showheap
fromelts i h g
showheap  
