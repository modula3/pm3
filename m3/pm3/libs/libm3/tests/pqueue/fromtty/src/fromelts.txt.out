> fromelts
> showheap
LAST(pq.heap^) = 0; pq.sz = 0
pq.heap = Elts{NIL}
> fromelts e d c b a z y x w
"e" => 1
"d" => 2
"c" => 3
"b" => 4
"a" => 5
"z" => 6
"y" => 7
"x" => 8
"w" => 9
> showheap
LAST(pq.heap^) = 9; pq.sz = 9
pq.heap = Elts{NIL, "a", "b", "c", "e", "d", "z", "y", "x", "w"}
> fromelts i h g
"i" => 10
"h" => 11
"g" => 12
> showheap  
LAST(pq.heap^) = 9; pq.sz = 3
pq.heap = Elts{NIL, "g", "h", "i", NIL (X6)}
> ^D
